### ISSUE TITLE
False-positive error when returning in async block

### ISSUE THREAD

**Issue by Seeker14491** [intellij-rust/intellij-rust#3870]
The following code is valid Rust, but in the IDE the `return` keyword shows an error:

```rust
#![feature(async_await)]

pub fn foo() -> i32 {
    let _ = async {
        return; // `return;` in a function whose return type is not `()`
    };

    0
}
```

## Environment

* **Intellij-Rust plugin version:** 0.2.97.4975-191
* **Rust toolchain version:** rustc 1.36.0-nightly (7d5aa4332 2019-05-16)
* **IDE name and version:** CLion 2019.1.2
* **Operating system:** Windows 10 x64

**Comment by rrevenantt** [comment]
Are you sure that it should be valid Rust? [RFC](https://github.com/rust-lang/rfcs/blob/master/text/2394-async_await.md#async-blocks) explicitly says that return is forbidden in async blocks

**Comment by Seeker14491** [comment]
Although it does state that at the top, under "Unresolved questions" at the bottom of that RFC it says "This RFC does not propose how the `?`-operator and control-flow constructs like return, break and continue should work inside async blocks". So it seems the RFC is undecided.
